docs/source/tutorial/modules.rst

```diff
@@ -74,14 +74,17 @@ The ``with`` keyword in expressions comes in two variants:
 This is particularly useful with ``do`` notation, where it can often improve
 error messages: ``with MyModule.(>>=) do ...``
 
-There is no formal link between the module name and its filename,
-although it is generally advisable to use the same name for each. An
-``import`` statement refers to a filename, using dots to separate
-directories. For example, ``import foo.bar`` would import the file
-``foo/bar.idr``, which would conventionally have the module declaration
-``module foo.bar``. The only requirement for module names is that the
-main module, with the ``main`` function, must be called
-``Main`` — although its filename need not be ``Main.idr``.
+If a file contains a module declaration ``module Foo.Bar.MyModule``, its
+path relative to the ``sourcedir`` specified in the ``.ipkg`` project file
+(defaults to `.`) must be ``./Foo/Bar/MyModule.idr``. If you are not using an
+``.ipkg`` project file, the path must be relative to the directory you are
+running Idris from. Similarly, an ``import`` statement also refers to such a
+relative filepath stripped of its file extension, using dots to separate
+directories. As in the example above, all modules names and directories must be
+capitalised identifiers. The main module, with the ``main`` function, must be
+called ``Main`` — although its filename need not be ``Main.idr``. This is the
```

Comment:
Sorry, I found this only now. There is not requirement that the main module is called `Main`. Again, in `ipkg` file you can set any module to be the main one with the `main` property.

Reply:
But if module does not have any `module <Name>` declaration in the beginning it indeed automatically gets its name to be `Main`. But it does not have to in order to be the starting point of an application

Reply:
No problem! I mentioned what happens to modules with no `module <Name>` declaration. I did not add the part about the `main` property for `.ipkg` files yet. Should I?

Reply:
> I did not add the part about the `main` property for `.ipkg` files yet. Should I?

Maybe. What I mean is that phrase

> The main module, with the ``main`` function, must be called ``Main``

is not correct

Reply:
Yep, I feel the same way about it, too.